### ISSUE TITLE
[Enhancement]: Multi-User Management and Account Expiration

### ISSUE THREAD

**Issue by LuckyUnderdog** [advplyr/audiobookshelf#2066]
### Describe the feature/enhancement

Dear Developers

Thank you for the incredible work you've done on the this project. After using it, I noticed that while the platform offers Multi-user support, it currently lacks some Multi-user management features. Many administrative tasks, such as disabling user accounts, need to be done manually.

If administrators able to set an expiration date for user accounts/disable user accounts base on time. This feature would be incredibly useful for scenarios such as trial accounts or temporary tests.

Additionaly, the ability to perform batch operations on user accounts and other administrative functions would be highly appreciated.

**Comment by nichwall** [comment]
I think something like Calibre-Web's user view would be great. This provides a good overview of everyone's permissions.

![calibre_web_summary](https://github.com/advplyr/audiobookshelf/assets/5686638/eaf9ef37-4dae-4af6-bc65-a753d081191c)

Clicking on an individual user opens up their normal user edit page (pop-up in ABS's case), or allows for batch editing using the "Edit Users" button to multi-select users on the left. When users are selected, the "Allow" and "Deny" is no longer greyed out at the top of every column to change all of their checkboxes at once. My only complaint with this is the table is very wide in the edit view (and we don't have nearly as many permissions) and changes are applied immediately. Implementing something similar to their table view would also make it easier to expand when new permissions are added.

![bulk_edit_page1](https://github.com/advplyr/audiobookshelf/assets/5686638/88c72288-fb63-40d2-b9d5-242ac6326182)
![bulk_edit_page2](https://github.com/advplyr/audiobookshelf/assets/5686638/ce90d2b4-075a-4e18-b59b-4fb4def703e5)


In the edit view, I would replace the checkboxes with the current sliders we already have. I think it would be nifty to have the slider color change like below, but absolutely unnecessary:
- green  : enabled and saved
- yellow : enabled and not saved
- red    : disabled and not saved
- gray   : disabled and saved

I'm not sure the best way to do batch editing of access control (to libraries and tags). Calibre-Web's works well, just feel like there's a better way.

**Comment by zkte** [comment]
yes with paypal integration and credit system would be perfect 👍
